NextcloudTalk/SettingsTableViewController.swift

```diff
@@ -0,0 +1,699 @@
+
+        switch settingsSection {
+        case SettingsSection.kSettingsSectionUserStatus.rawValue:
+            return 1
+        case SettingsSection.kSettingsSectionConfiguration.rawValue:
+            return getConfigurationSectionOptions().count
+        case SettingsSection.kSettingsSectionAbout.rawValue:
+            return AboutSection.kAboutSectionNumber.rawValue
+        case SettingsSection.kSettingsSectionAccounts.rawValue:
+            return NCDatabaseManager.sharedInstance().inactiveAccounts().count
+        default:
+            break
+        }
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        let sections = getSettingsSections()
+        let currentSection = sections[indexPath.section]
+        if currentSection == SettingsSection.kSettingsSectionUser.rawValue {
+            return 100
+        }
+        return 48
+    }
+
+    override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        let sections = getSettingsSections()
+        let settingsSection = sections[section]
+
+        switch settingsSection {
+        case SettingsSection.kSettingsSectionUserStatus.rawValue:
+            return NSLocalizedString("Status", comment: "")
+        case SettingsSection.kSettingsSectionAccounts.rawValue:
+            return NSLocalizedString("Accounts", comment: "")
+        case SettingsSection.kSettingsSectionConfiguration.rawValue:
+            return NSLocalizedString("Configuration", comment: "")
+        case SettingsSection.kSettingsSectionAbout.rawValue:
+            return NSLocalizedString("About", comment: "")
+        default:
+            break
+        }
+        return nil
+    }
+
+    override func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        let sections = getSettingsSections()
+        let settingsSection = sections[section]
+        if settingsSection == SettingsSection.kSettingsSectionAbout.rawValue {
+            let appName = (Bundle.main.object(forInfoDictionaryKey: "CFBundleDisplayName") as? String)!
+            let appVersion = (Bundle.main.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String)!
+            return appName + appVersion + copyright
+        }
+        if let activeUserStatus = activeUserStatus {
+            if settingsSection == SettingsSection.kSettingsSectionUserStatus.rawValue && activeUserStatus.status == kUserStatusDND {
+                return NSLocalizedString("All notifications are muted", comment: "")
+            }
+        }
+        if let contactSyncSwitch = contactSyncSwitch {
+            if settingsSection == SettingsSection.kSettingsSectionConfiguration.rawValue && contactSyncSwitch.isOn {
+                if NCContactsManager.sharedInstance().isContactAccessDetermined() && NCContactsManager.sharedInstance().isContactAccessAuthorized() {
```

Comment:
The check should be
`if NCContactsManager.sharedInstance().isContactAccessDetermined() && !NCContactsManager.sharedInstance().isContactAccessAuthorized()`